File: camerax_android/lib/src/legacy/video/recording_stats.dart
import 'package:camerax_android/src/legacy/camerax.g.dart' as $native;
import 'package:camerax_platform_interface/camerax_platform_interface.dart'
    as $interface;

import 'audio_stats.dart';

extension RecordingStatsObj on $native.RecordingStats {
  $interface.RecordingStats get args {
    return $interface.RecordingStats(
      audioStats: audioStats.args,
      numBytesRecorded: numBytesRecorded,
      recordedDurationNanos: recordedDurationNanos,
    );
  }
}

File: camerax_android/android/src/main/kotlin/dev/hebei/camerax_android/legacy/video/RecordingStats.kt
package dev.hebei.camerax_android.legacy.video

import dev.hebei.camerax_android.legacy.CameraXRegistrar
import dev.hebei.camerax_android.legacy.PigeonApiRecordingStats

class RecordingStats(registrar: CameraXRegistrar) : PigeonApiRecordingStats(registrar) {
    override fun audioStats(pigeon_instance: androidx.camera.video.RecordingStats): androidx.camera.video.AudioStats {
        return pigeon_instance.audioStats
    }

    override fun numBytesRecorded(pigeon_instance: androidx.camera.video.RecordingStats): Long {
        return pigeon_instance.numBytesRecorded
    }

    override fun recordedDurationNanos(pigeon_instance: androidx.camera.video.RecordingStats): Long {
        return pigeon_instance.recordedDurationNanos
    }
}